src/Nethermind/Nethermind.Network/P2P/ProtocolHandlers/ProtocolHandlerBase.cs

```diff
@@ -59,7 +59,14 @@ protected ProtocolHandlerBase(ISession session, INodeStatsManager nodeStats, IMe
         {
             try
             {
-                return _serializer.Deserialize<T>(data);
+                int originalReaderIndex = data.ReaderIndex;
+                T result = _serializer.Deserialize<T>(data);
+                if (data.IsReadable())
+                {
+                    throw new IncompleteDeserializationException(
```

Comment:
What does that exception achieve?

Reply:
Make it easier to detect deserialization issue.